docs/proposals/dedicated-nodes.md

```diff
@@ -0,0 +1,294 @@
+
+We will add
+
+```go
+	Tolerations []Toleration  `json:"tolerations,omitempty"`
+```
+
+to `PodSpec`. A pod must tolerate all of a node's taints (except taints
+of type TaintEffectPreferNoSchedule) in order to be able
+to schedule onto that node.
+
+We will add
+
+```go
+	Taints []Taint  `json:"hardTaints,omitempty"`
+```
+
+to both `NodeSpec` and `NodeStatus`. The value in `NodeStatus` is the union
+of the taints specified by various sources. For now, the only source is
+the `NodeSpec` itself, but in the future one could imagine a node inheriting
+taints from pods (if we were to allow taints to be attached to pods), from
+the node's startup coniguration, etc. The scheduler should look at the `Taints`
+in `NodeStatus`, not in `NodeSpec`.
+
+Taints and tolerations are not scoped to namespace.
+
+## Implementation plan: taints, tolerations, and dedicated nodes
+
+Using taints and tolerations to implement dedicated nodes requires these steps:
+
+1. Add the API described above
+1. Add a scheduler predicate function that respects taints and tolerations (for TaintEffectNoSchedule)
+and a scheduler priority function that respects taints and tolerations (for TaintEffectPreferNoSchedule). 
+1. Add to the Kubelet code to implement the "no admit" behavior of TaintEffectNoScheduleNoAdmit and
+TaintEffectNoScheduleNoAdmitNoExecute, and the eviction behavior of TaintEffectNoScheduleNoAdmitNoExecute.
+Remember that eviction may need to happen under a variety of circumstances: when a taint is added,
+when an existing taint is updated, when a toleration
+is removed from a pod, or when a toleration is modified on a pod. Note that we need a
+similar mechanism for changes that asynchronously trigger evictions due to hard affinity
+becoming violated (#18621 and #18625).
+1. Add a new `kubectl` command that adds/removes taints to/from nodes,
+1. (This is the one step is that is specific to dedicated nodes)
+Implement an admission controller that adds tolerations to pods that are supposed
+to be allowed to use dedicated nodes (for example, based on pod's namespace).
+
+In the future one can imagine a generic policy configuration that configures
+an admission controller to apply the appropriate tolerations to the desired class of pods and
+configures the NodeController or another component to apply the appropriate
+taints to the desired set of nodes. It could be used not just for policies about dedicated nodes,
+but also other uses of taints and tolerations, e.g. nodes that are restricted
+due to their hardware configuration.
+
+The `kubectl` command to add and remove taints on nodes will be modeled after `kubectl label`.
+Examples usages:
+
+```sh
+// Update node 'foo' with a taint with key 'dedicated' and value 'special-user' and effect 'NoScheduleNoAdmitNoExecute'.
+// If a taint with that key already exists, its value and effect are replaced as specified.
+$ kubectl taint nodes foo dedicated=special-user=NoScheduleNoAdmitNoExecute
```

Comment:
How about `dedicated=special-user:NoScheduleNoAdmitNoExecute`. (Instead of a second `=`)